src/frontend/screens/Settings/MapSettings/BackgroundMapInfo.tsx

```diff
@@ -0,0 +1,330 @@
+  },
+  lvlOfDetail17: {
+    id: "screens.Settings.MapSettings.BackgroundMapInfo.lvlOfDetail17",
+    defaultMessage: "Block, park, addresses",
+  },
+  lvlOfDetail18: {
+    id: "screens.Settings.MapSettings.BackgroundMapInfo.lvlOfDetail18",
+    defaultMessage: "Some buildings, trees",
+  },
+  lvlOfDetail19: {
+    id: "screens.Settings.MapSettings.BackgroundMapInfo.lvlOfDetail19",
+    defaultMessage: "Local highway and crossing details",
+  },
+  lvlOfDetail20: {
+    id: "screens.Settings.MapSettings.BackgroundMapInfo.lvlOfDetail20",
+    defaultMessage: "A mid-sized building",
+  },
+});
+
+// To Do: Get level of detail from programs team
+export const BackgroundMapInfo = ({
+  route,
+  navigation,
+}: NativeRootNavigationProps<"BackgroundMapInfo">) => {
+  const { formatMessage: t } = useIntl();
+  const { bytesStored, id, styleUrl, name } = route.params;
+  const [zoom, setZoom] = React.useState<"loading" | number | null>("loading");
+
+  const sheetRef = React.useRef<BottomSheetMethods>(null);
+
+  const levelOfDetail = React.useMemo(() => {
+    if (typeof zoom !== "number") return undefined;
+
+    switch (true) {
+      case zoom >= 0 && zoom < 2:
+        return m.lvlOfDetail0to1;
+      case zoom === 2:
+        return m.lvlOfDetail2;
+      case zoom >= 3 && zoom < 5:
+        return m.lvlOfDetail3to4;
+      case zoom === 5:
+        return m.lvlOfDetail5;
+      case zoom === 6:
+        return m.lvlOfDetail6;
+      case zoom >= 7 && zoom < 9:
+        return m.lvlOfDetail7to8;
+      case zoom === 9:
+        return m.lvlOfDetail9;
+      case zoom === 10:
+        return m.lvlOfDetail10;
+      case zoom === 11:
+        return m.lvlOfDetail11;
+      case zoom === 12:
+        return m.lvlOfDetail12;
+      case zoom >= 13 && zoom < 15:
+        return m.lvlOfDetail13to14;
+      case zoom === 15:
+        return m.lvlOfDetail15;
+      case zoom === 16:
+        return m.lvlOfDetail16;
+      case zoom === 17:
+        return m.lvlOfDetail17;
+      case zoom === 18:
+        return m.lvlOfDetail18;
+      case zoom === 19:
+        return m.lvlOfDetail19;
+      case zoom === 20:
+        return m.lvlOfDetail20;
+      default:
+        return undefined;
+    }
+  }, [zoom]);
+
+  const { setStyleId } = useMapStyle();
+
+  function setStyleAndNavigateHome() {
+    setStyleId(id);
+    navigation.navigate("Home", { screen: "Map" });
+  }
+
+  React.useEffect(() => {
+    api.maps
+      .getStyle(id)
+      .then(style => {
+        const tileSetIdArray = Object.values(style.sources).map(source => {
+          if ("url" in source && source.url) {
+            const url = source.url;
+            const searchWord = "tilesets/";
+            const startPosition = url.search(searchWord);
+            if (startPosition === -1) return;
+            const endPosition = url.length;
+            return url.substring(
+              startPosition + searchWord.length,
+              endPosition
+            );
+          }
```

Comment:
seems like this will work but a little hard to parse. not sure if this is easier to read but would lean on using the `URL` object, which would also confirm that the url is a proper url:

```js
const url = new URL(source.url)

if (url.pathname.startsWith('/tilesets/')) {
  const splitPathname = url.pathname.split("/")
  return splitPathname[splitPathname.length - 1]
}
```